src/robot/conf/settings.py

```diff
@@ -92,6 +94,9 @@ def __setitem__(self, name, value):
     def _process_value(self, name, value):
         if name == 'ReRunFailed':
             return gather_failed_tests(value)
+        if name == 'ReRunFailedSuites':
+            # will be calculated at the end
+            return lambda: gather_failed_suites(value, self.critical_tags, self.non_critical_tags)
```

Comment:
Returning a lambda that is called in `_process_cli_opts()` is a bit ugly but works. Alternatively we could just let the normal logic set `ReRunFailedSuites` option and in `_process_cli_opts()` do something like this:

```python
if self['ReRunFailedSuites']:
    self['SuiteNames'] += gather_failued_suites(self['ReRunFailedSuites'])
```

For consistency reasons the logic with `ReRunFailed` should be changed to follow this logic too.

What do you think?